src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,98 @@
-export const Pagination = () => {};
+import classNames from 'classnames';
+import React from 'react';
+
+type PaginationProps = {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+};
+
+export const Pagination: React.FC<PaginationProps> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const arrayOfPages = Array.from(
+    { length: Math.ceil(total / perPage) },
+    (_, i) => i + 1,
+  );
+
+  const arrayOfItems = Array.from(
+    { length: Math.min(perPage, total - (currentPage - 1) * perPage) },
+    (_, i) => {
+      return `Item ${i + 1 + (currentPage - 1) * perPage}`;
+    },
+  );
+
+  return (
+    <div>
+      <ul className="pagination">
+        <li
+          className={classNames(`page-item`, { disabled: currentPage === 1 })}
+        >
+          <a
+            data-cy="prevLink"
+            className="page-link"
+            href="#prev"
```

Comment:
Using `href="#prev"` can cause unwanted behavior in the application by updating the URL hash. It's better to prevent the default action in the `onClick` event handler to avoid this.